TaskHero/ParseClient.swift

```diff
@@ -55,7 +54,48 @@ class ParseClient: NSObject {
         t["name"] = task.name
         t["details"] = task.details
         t["estimated_time"] = task.estimatedTime
+        
+        // TODO: Generate Task ID
+    
+        if let steps = task.steps {
+            let stepsData = steps.map({ (step) -> [String : AnyObject] in
+                var assigneeIds: [String] = []
+                if let assignees = step.assignees {
+                    assigneeIds = assignees.map { user in user.id! }
+                }
+                
+                return [
+                    "name": step.name as AnyObject,
+                    "details": step.details as AnyObject,
+                    "assignees": assigneeIds as AnyObject
+                ]
+            })
+            
+            let data = try! JSONSerialization.data(withJSONObject: stepsData, options: [])
+            t["steps"] = String(data: data, encoding: .utf8)
+        }
+        
+        t.saveInBackground { (saved, error) in
+            if let error = error {
+                failure(error)
+            } else {
+                success()
+            }
+        }
+    }
     
+    func createTaskInstance(task: Task, success: @escaping () -> (), failure: @escaping (Error) -> ()) {
+        let u = PFUser()
+        u.objectId = User.current!.id
+        
+        let t = PFObject(className: "TaskInstances")
+        t["author"] = u
+        t["name"] = task.name
+        t["details"] = task.details
+        t["estimated_time"] = task.estimatedTime
```

Comment:
So I think we'll need a few more fields here - comparing it to our TaskIntance.json meta file, we also want a "task_id" and a "chat_id".

For now, I'd just set the "chat_id" to some arbitrary value, but the "task_id" should be a pointer to the Task from which this was derived from. So I believe you'll want to set it to the actual PFObject (similar to how the "author" field is set to the PFUser object).